src/history/hs.c

```diff
@@ -829,179 +831,166 @@ __wt_find_hs_upd(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE **upd
      */
     read_timestamp = allow_prepare ? txn->prepare_timestamp : txn->read_timestamp;
     ret = __wt_hs_cursor_position(session, hs_cursor, hs_btree_id, key, read_timestamp);
```

Comment:
The diff doesn't look nice here. Essentially we're no longer checking visibility with the key information and we're instead inferring visibility from the fact that we were able to see the record in the first place so there is no need to run this code in a loop.

To make the smallest change possible, I've brought the logic out of the loop and each spot where we've used `break` to zip to the end of the function to do the cleanup, I've added `goto done` to replicate this control flow.